lib/elserver/handler.ex

```diff
@@ -8,8 +8,13 @@ defmodule Elserver.Handler do
   end
 
   def parse(request) do
-    # TODO: Parse the request string into a map:
-    conv = %{ method: "GET", path: "/wildthings", resp_body: "" }
+    #Parse the request string into a map:
+    [method, path, _] = 
+      request 
+      |> String.split("\n") 
+      |> List.first
+      |> String.split(" ")
+    %{ method: method, path: path, resp_body: "" } #last expression of a variable is implicitly returned
   end
```

Comment:
Here we use the parse function to split the request at every new  and the we pattern match the request sections into a list of items and we return a map filled with the bits we need from the request in order to route the request